core/src/main/scala/com/rackspace/com/papi/components/checker/util/Instrumented.scala

```diff
@@ -0,0 +1,62 @@
+package com.rackspace.com.papi.components.checker.util
+
+import com.codahale.metrics.{Gauge, Metric, MetricFilter, MetricRegistry}
+import com.rackspace.com.papi.components.checker.Validator
+import com.typesafe.scalalogging.slf4j.LazyLogging
+
+/**
+  * @see nl.grons.metrics.scala.InstrumentedBuilder
+  */
+trait Instrumented extends LazyLogging {
+  /** The base name for all metrics created from this builder. */
+  val metricBaseName = Validator.metricDomain
+
+  /**
+    * The MetricRegistry where created metrics are registered.
+    */
+  val metricRegistry: MetricRegistry = Validator.metricRegistry
+
+  def getRegistryClassName(clazz: Class[_]): String = {
+    clazz.getName
+      .replace(metricBaseName, "")
+      // @see nl.grons.metrics.scala.MetricName
+      .replaceAllLiterally("$$anonfun", ".")
+      .replaceAllLiterally("$apply", ".")
+      .replaceAll("""\$\d*""", ".")
+      .replaceAllLiterally(".package.", ".")
+      .replaceAll("""^\.""", "")
+      .replaceAll("""\.$""", "")
+  }
+
+  def guard[A](buildMetric: () => A): Option[A] = {
+    try {
+      Some(buildMetric())
+    } catch {
+      case e: IllegalArgumentException =>
+        logger.warn("Failed to add new metric. Reason: {}", e.getMessage)
```

Comment:
The latest update makes it so `guard()` can't be called outside of `gaugeOrAdd()`. The latter ensures that either a new `Gauge` was added or the existing one is returned. This is the behavior of the old libraries. So yes we still hit the exception, and yes we do return the already registered `Gauge`.